### ISSUE TITLE
Bancontact spike part II: Limonetik & Ingenico

### ISSUE THREAD

**Issue by RachL** [openfoodfoundation/openfoodnetwork#4432]
## Description

A first spike was run here https://github.com/openfoodfoundation/openfoodnetwork/issues/3901 and handed over to the product team.

The discussion went on in discourse around the scope and the strategy we shoudl apply IF we integrate another payment gateway.

Myriam did a deep comparison of providers and found a particular interesting offer at Limonetik but also at Ingenico.

She detailed everything here : https://docs.google.com/presentation/d/10dJjMwmLq274ecPByvhaCUUgUnNG7P7MUzRl_KkmBrA/edit#slide=id.p

Full discourse post: https://community.openfoodnetwork.org/t/bancontact-payment-method/1694/22

The purpose of this last issue is to understand if it is technically worth to pursue the Limonetik/Ingenico negotiation.

Some tech documentation might be needed: Rachel has the contacts.

## Acceptance Criteria & Tests

1. Provide T-Shirt size of the work needed to implement Limonetik's solution.
2.  If remaining time: do the same for Ingenico
3.  Provide recommendation from a tech point of vue of the best approach for OFN : stay with Stripe or add a second gateway

**Comment by RachL** [comment]
@sigmundpetersen I won't be doing anything, it is really up for a dev to work on it. So I will unassigned myself ;)

**Comment by sigmundpetersen** [comment]
Yeah I was thinking as product lead, but since this is just the spike I agree. Have we clarified if we assign prod leads to the feature epics?

**Comment by pacodelaluna** [comment]
@RachL @myriamboure I will take a deeper look at these 2 platforms, especially on technical integration, during the 2-3 next weeks.

**Comment by pacodelaluna** [comment]
@RachL @myriamboure @sigmundpetersen I will be glad to take a deeper look at these 2 platforms, especially on technical integration, during the 2-3 next weeks.

**Comment by RachL** [comment]
@sigmundpetersen I don't know :) We do it sometimes for epic.

@pacodelaluna Cool! It will be nice to have a second pair of eyes.

**Comment by daniellemoorhead** [comment]
Hey there @pacodelaluna and @RachL has this work been started? Does it go into the `in dev` column on zenhub? Or still to be done? :)

**Comment by RachL** [comment]
@daniellemoorhead Francois will put his findings here but we still need another dev to look at it. So it is still to be done, that's why I've left it in dev ready.

**Comment by pacodelaluna** [comment]
I will just pasted the quick investigation I made, took sometimes to read all the doc about it and get it all.

Source:
https://community.openfoodnetwork.org/t/bancontact-payment-method/1694/28
https://docs.google.com/presentation/d/10dJjMwmLq274ecPByvhaCUUgUnNG7P7MUzRl_KkmBrA/edit#slide=id.p
https://github.com/openfoodfoundation/openfoodnetwork/issues/3901

Ingenico
- https://epayments-api.developer-ingenico.com/s2sapi/v1/en_US/ruby/concepts.html?paymentPlatform=ALL
- https://github.com/Ingenico-ePayments/connect-sdk-ruby
- https://epayments.developer-ingenico.com/signup-for-sandbox/

Limonetik
- https://docs.limonetik.com/v1.0/reference
- No Ruby SDK, API quite clean
- Sandbox on request

From what I got, Ingenico technical solution looks stronger (more services and integration also) than Limonetik one (which is using Ingenico services at some levels ...), could be also easier to integrate thanks to their Ruby SDK.
This is just a technical insight, as from what I got, the issue is that Ingenico will ask a minimum amount to invoice in order to allow us to use their service, but I guess we can negotiate this a bit.

**Comment by luisramos0** [comment]
hello, I guess this was asking for a second dev to look at it. I had a quick look.

Re Estimations below: I am assuming the integration would just for the payment processes that exist right now in the ofn app: checkout, add card, capture payment in admin and subscriptions.
Also, I am not sure yet how much more difficult the SCA authentication flows will make these integrations, I dont think it will make them much longer.

1. T-Shirt size to implement Limonetik
Integrating a new gateway will be a L. Something like 2 weeks FTE for a dev.

2. T-Shirt size to implement Ingenico
The Ingenico sdk will make its integration something like 2 days less of dev compared to Limonetik.
Convert from php to ruby: https://github.com/winzou/limonetik-php-api

3. "stay with Stripe or add a second gateway"
It's harder to have multiple gateways but if we want to offer cheaper options to the hubs, we must do this. I think we can do it. The key point is really to select the right provider that will help us long term...

I am not sure I understood all the details of the comparison between Limonitek and Ingenico but I wonder if we could combine this initiative with the marketplace payments initiative and get the two problems solved in one go... I see mangopay fees are even higher than Stripe, I wonder what are the alternatives to mangopay and what are their fees.

**Comment by luisramos0** [comment]
both tech analysis of this issue are pretty quick/superficial. I wonder if we should investigate more or if we have enough tech info to make a decision.

**Comment by RachL** [comment]
@luisramos0 

> I am not sure I understood all the details of the comparison between Limonitek and Ingenico but I wonder if we could combine this initiative with the marketplace payments initiative and get the two problems solved in one go... I see mangopay fees are even higher than Stripe, I wonder what are the alternatives to mangopay and what are their fees.

That's the work that has been done mainly by Myriam on discourse. This is how we ended up with Limonetik and Ingenico : those are the cheapest solution we could bring for the hubs. Mangopay / Stripe & like have pretty much the same prices.

Is it correct to draw the following conclusions:

- Adding a new payment gateway is more expensive, but we may have business reasons to do it
- We need now to prioritize needs (Bancontact, cheaper solution for hubs, split payments....)
- Once we will have a priority that impacts the payment methods we will need to make the decision whether or not we take the opportunity to add another gateway at the same time. The choice between Limonetik or Ingenico will mostly remain on the product/business side as there is not much difference on the tech side.

If we are aligned, I will put this into discourse and close the spike :-)

**Comment by luisramos0** [comment]
yeah, agree, thanks for clarifying!

I dont think these new gateways will help with the "split payments" problem...
